tabrepo/repository/abstract_repository.py

```diff
@@ -532,3 +560,126 @@ def _convert_binary_to_multiclass(self, predictions: np.ndarray, dataset: str) -
             return np.stack([1 - predictions, predictions], axis=predictions.ndim)
         else:
             return predictions
+
+    # TODO: repo time_infer_s is per row, results_df is the total time for all rows, need to align later
+    # TODO: Error if unknown configs/baselines requested
+    # TODO: Add fillna
+    # TODO: Docstring
+    # Q:Whether to keep these functions a part of TabRepo or keep them separate as a part of new fit()-package
+    def compare_metrics(
```

Comment:
Oh I forgot to make this point in my review. I would highly recommend to move `compare_metrics` and `plot_overall_rank_comparison` into utils and out of repository base class as those methods are highly complex (they double the LOC of the class) and only specific to one use-case.

Reply:
Agreed, although I'd prefer to do this in a follow-up PR. I think I'll keep it as is in this PR and make a dedicated PR to move this logic so it is easier to review.

These two methods are WIP, so I'm planning for them to change quite a bit before 2.0.